pkg/util/sync_test.go

```diff
@@ -0,0 +1,44 @@
+package util
+
+import (
+	"fmt"
+)
+
+func ExampleProduceMapReduce() {
```

Comment:
Can we make this a real test?

Reply:
It is a real test:

```
$ go test -v ./pkg/util | grep Example
=== RUN   ExampleProduceMapReduce
--- PASS: ExampleProduceMapReduce (0.00s)
```

and is [supposed to be how examples are embedded in package documentation](https://go.dev/blog/examples), which is why I decided to try to write it this way.

Reply:
After the usual fight (I hate Go modules), this is how it looks like rendered:

<details>

![Screen Shot 2022-02-09 at 12 50 29](https://user-images.githubusercontent.com/934092/153204813-544186ad-9aea-4248-8289-8ec34b9b3fce.png)

</details>

Apparently, there is no way to format `code`, ` `` ` is not respected and I can't find examples in the stdlib.

Reply:
TIL

Reply:
Pretty awesome